scripts/make_report_tables.py

```diff
@@ -0,0 +1,777 @@
+    if not os.path.isdir(results_folder):
+        os.mkdir(results_folder)
+
+    # Save
+    with open(
+        os.path.join(results_folder, "sim_correlations_" + tname + ".tex"), "w"
+    ) as f:
+        f.write(tex_table)
+
+    # No Return Value
+    return None
+
+
+# === H4 Functions === #
+
+# H4: Dataset size (or ratios) and transfer success
+# 2 directions of attack
+# 2 distributions attacks are based on
+# 2 kinds of attack
+# 2 transfer success metrics
+# 4+ versions (target size, surrogate size, t/s ratio, s/t ratio)
+# = lots of possible ways of doing this
+
+TYPES = ["ratio", "diff", "abs-diff"]
+TYPE_LABELS = ["Ratio", "Difference", "Absolute Difference"]
+
+
+def make_h4_cor_table(
+    wandb_runs: wandb.Api.runs,
+    directions: list[str],
+    distributions: list[str],
+    atk_name: str,
+    atk_metric_names: str,
+    type: str,
+) -> list[list[pearsonr]]:
+    # Output Array: similarity metrics on rows, attack metrics on columns
+    out = []
+
+    # Counter
+    i = 0
+
+    # Read in all dataset dicts to extract sizes
+    experiment_groups_file_names = os.listdir(constants.EXPERIMENT_GROUPS_PATH)
+    experiment_groups = {}
+    for eg_file_name in experiment_groups_file_names:
+        with open(
+            os.path.join(
+                constants.EXPERIMENT_GROUPS_PATH, experiment_groups_file_names[0]
+            )
+        ) as file:
+            experiment_groups[eg_file_name[:-5]] = yaml.safe_load(file)
+
+    # To access dmpair 0 of little-blur experiment group:
+    # experiment_groups["little-blur"]["dmpairs"][0]
+
+    # Loop over directions (A2B, B2A)
+    for direction in directions:
+
+        # Filter down runs
+        runs = [run for run in wandb_runs if direction in run.summary.keys()]
+
+        # Loop over distributions (A, B)
+        for distribution in distributions:
+
+            # New column in output
+            out.append([])
+
+            # Get dataset drop ratios
+            drop_scores = []
+            for run in runs:
+                # Get experiment group name, dmpair num, and surrogate letter
+                run_eg = run.name[:-6]
+                dmpair_num = int(run.name[-5])
+                surrogate = run.name[-1]
+
+                # Get corresponding dataset sizes
+                drops = experiment_groups[run_eg]["dmpairs"][dmpair_num]
+                A_size = CIFAR_SIZE * (1 - drops["A"]["drop"])
+                B_size = CIFAR_SIZE * (1 - drops["B"]["drop"])
+
+                # Get surrogate and target drop score depending on type
+                if type == "ratio":
+                    # surrogate/target drop ratio
+                    if surrogate == "A":
+                        drop_scores.append(B_size / A_size)
+                    if surrogate == "B":
+                        drop_scores.append(A_size / B_size)
+                if type == "diff":
+                    # surrogate/target drop ratio
```

Comment:
Small one - this comment doesn't refer to the 'diff' type

Reply:
Have fixed